src/amo/components/CategoryInfo.js

```diff
@@ -0,0 +1,64 @@
+import classNames from 'classnames';
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { asyncConnect } from 'redux-connect';
+import { compose } from 'redux';
+
+import translate from 'core/i18n/translate';
+import { loadCategoriesIfNeeded } from 'core/utils';
+
+import './CategoryInfo.scss';
+
+
+export class CategoryInfoBase extends React.Component {
+  static propTypes = {
+    addonType: PropTypes.string.isRequired,
+    categories: PropTypes.arrayOf(PropTypes.object),
+    clientApp: PropTypes.string.isRequired,
+    slug: PropTypes.string.isRequired,
+  }
+
+  render() {
+    const { addonType, categories, clientApp, slug } = this.props;
+
+    if (!categories) {
+      return null;
+    }
+
+    const categoryMatch = categories.filter((category) => {
+      return (category.application === clientApp && category.slug === slug &&
+        category.type === addonType);
+    });
+    const category = categoryMatch.length ? categoryMatch[0] : false;
```

Comment:
There's an [`Array.prototype.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) that would remove the need for this but restructuring how we store categories is probably better.